src/Framework/Performance/Measurement.cxx

```diff
@@ -12,13 +12,17 @@ static long unsigned int get_current_time() {
   return system_clock::to_time_t(current_time);
 }
 
-Measurement::Measurement(bool do_sample) {
-  if (do_sample) sample();
+Measurement::Measurement(bool do_start) {
+  if (do_start) start();
 }
 
-void Measurement::sample() {
+void Measurement::start() {
+  start_ = get_current_time();
+}
+
+void Measurement::end() {
   valid_ = true;
-  time_ = get_current_time();
+  duration_ = get_current_time() - start_;
```

Comment:
I'm not 100% sure this will do anything sensible, since start_/get_current_time are std::time_t objects. I don't think subtracting them gives a valid duration. 

I would make start_ a std::chrono time_point and have some using declarations in the header that determine 
- What clock to use (e.g. `using clock = std::chrono::system_clock` `using time_point = clock::time_point`
- What duration to save (potentially something you could customize later?)
`using duration = std::chrono::nanoseconds` 

Then duration_ woudl be recorded as 
```
std::chrono::duration_cast<duration>(get_current_time() - start_).count()
```

or something like this

Reply:
Yea, I'm doing another rewrite where I use chrono duration so I can get the seconds as a `double`.